pkg/types/status.go

```diff
@@ -58,7 +58,7 @@ func (s *Status) Index() int {
 	return int(*s)
 }
 
-func (s *Status) MarshalJSON() ([]byte, error) {
+func (s Status) MarshalJSON() ([]byte, error) {
```

Comment:
Now struct Status has methods on both value (`UnmarshalJSON`) and pointer (`String`, `Index` and `UnmarshalJSON`) receivers.

Should we change receivers for other functions too?

Reply:
But in this case we can't set Status in `UnmarshalJSON`, can we?

Reply:
it seems it's a common practice to use `value` receiver for UnmarshalJSON and `pointer` for other methods:
https://github.com/CycloneDX/cyclonedx-go/blob/6f53207eded10e9e2a362772b253621840e7ad94/cyclonedx_json.go

Reply:
in the docs
https://pkg.go.dev/encoding/json#example-package-CustomMarshalJSON

Reply:
Yes, MarshalJSON and UnmarshalJSON should be excluded from the linter rule.
https://youtrack.jetbrains.com/issue/GO-13587